### PR TITLE
Week 3 assessment y

### PR DESCRIPTION
**Pull request by yanxu2021**
Complete the assessment, please review and approve the pull request.

Thank you, 

Sincerely,

Yan

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,50 +16,168 @@
...
+    }else{
+        let arr = []
+        arr[0] = 1
+        arr[1] = 1
+        for(let i = 2; i < num; i++){
+            arr[i] = arr[i-1] + arr[i-2]
+        }
+        return arr
+    }
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:62]: Great Work!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,50 +16,168 @@
...
+    // Check if the variable is an array, using the isArray method.
+    if(isArray(arr)){
+        // return arr.filter(value => (typeof value == `number` && value % 2 !== 0))--->Output: [ 9, 7, 199, -9 ]
+        // return (arr.filter(value => (typeof value == `number` && value % 2 !== 0))).sort()--->Output: [ -9, 199, 7, 9 ]   
+        // Numeric Sort: by default, the .sort() sorts values as strings.
+        // Get numeric sort done by: .sort(function(a,b){return a-b})
+        return (arr.filter(value => (typeof value == `number` && value % 2 !== 0))).sort(function(a,b){return a-b})
+        //Output: [ -9, 7, 9, 199 ]
+    }else{
+        return `Please input an array.`
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:99]: Good thinking about the edge case.

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,50 +16,168 @@
...
+            //An empty array should return an empty array.
+        }
+        return newArr
+    }else{
+        return `Please input an array.`
+    }
+}
+        
+console.log(accSum1([1,3,5]))// Expected output: [1,4,9]
+console.log(accSum1(`yan`))
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:154]: Great Job!

File: interview-questions.md [hunk 4]
```diff
@@ -9,59 +9,91 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-  Researched answer:
+  Your answer: The key word 'this' is used as a reference to the current object or parameter.//I first learned 'this' in C#, which refers to the current instance of the class and is also used as a modifier of the first parameter of an extension method. In javaScript I didn't pay attention to details. Yes, 'this' confused me now and I know we should use 'this' in the content but actually sometimes I don't know why. 
 
+  Researched answer: The JavaScript 'this' keyword refers to the object it belongs to. It has different values depending on where it is used:
+  - In a method, 'this' refers to the owner object.
+  - Alone, 'this' refers to the global object.
+  - In a function, 'this' refers to the global object.
+  - In a function, in strict mode, 'this' is 'undefined'.
+  - In an event, 'this' refers to the element that received the event.
+  - Methods like call(), and apply() can refer 'this' to any object.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:20]: "This" can get quite confusing depending on its context.

File: interview-questions.md [hunk 5]
```diff
@@ -9,59 +9,91 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: 'yarn' is package manager and it helps us to use exist/premade code/package to develop. 
 
+  Researched answer:Yarn is a package manager for JavaScript, which aims to be speedy, deterministic, secure and allows the user to use and share code with other develpers.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:27]: Great answer!

File: interview-questions.md [hunk 6]
```diff
@@ -9,59 +9,91 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  - The following shows an anonymous function that displays a message:
+
+  ```JavaScript
+  let show = function () {
+      console.log('Anonymous function');
+  };
+  show();
+  ```
+  - In this example, the anonymous function has no name between the function keyword and parentheses ().
+  - Assign the function to the show variable so that later the anonymous function can be called.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:73]: Good!

File: react-debugging.html [hunk 7]
```diff
@@ -79,10 +80,10 @@ <h1>Draw a Card</h1>
           }
           return(
             <>
-              <button onClick={}>
+              <button onClick={this.props.drawCard}>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:83]: Well done!

**Review by WalkerTexas-coder — APPROVED** [review]
Amazing work Yan! Thank you for putting in the research on the interview questions. Your code is highly readable and your logic is sound.  The React Debugging seemed to be no issue for you. Well done. Your PR is approved!
